CodeApp/Managers/MainApp.swift

```diff
@@ -328,9 +328,27 @@ class MainApp: ObservableObject {
             UUID().uuidString)
         try data.write(to: tempFile)
         var encoding: String.Encoding = .utf8
-        let fileContent = try String(contentsOf: tempFile, usedEncoding: &encoding)
-        try FileManager.default.removeItem(at: tempFile)
-        return (fileContent, encoding)
+        do{
+            let fileContent = try String(contentsOf: tempFile, usedEncoding: &encoding)
+            try FileManager.default.removeItem(at: tempFile)
+            return (fileContent, encoding)
+        }catch{
+            do{
+                let fileContent = try String(contentsOf: tempFile, encoding: .gb_18030_2000)
+                encoding = .gb_18030_2000
+                try FileManager.default.removeItem(at: tempFile)
+                return (fileContent, encoding)
+
+            }catch{
+              
+                    let fileContent = try String(contentsOf: tempFile, usedEncoding: &encoding)
+                    try FileManager.default.removeItem(at: tempFile)
+                    return (fileContent, encoding)                }
```

Comment:
This block of code is redundant, since it has already failed in the first trial. Instead of trying to catch the error while reading the file as GB, we can directly throw the error.